sentry-android-replay/src/main/java/io/sentry/android/replay/capture/BufferCaptureStrategy.kt

```diff
@@ -0,0 +1,174 @@
+    private val random: SecureRandom,
+    replayCacheProvider: ((replayId: SentryId) -> ReplayCache)? = null
+) : BaseCaptureStrategy(options, hub, dateProvider, recorderConfig, replayCacheProvider = replayCacheProvider) {
+
+    private val bufferedSegments = mutableListOf<ReplaySegment.Created>()
+
+    internal companion object {
+        private const val TAG = "BufferCaptureStrategy"
+    }
+
+    override fun stop() {
+        val replayCacheDir = cache?.replayCacheDir
+        replayExecutor.submitSafely(options, "$TAG.stop") {
+            FileUtils.deleteRecursively(replayCacheDir)
+        }
+        super.stop()
+    }
+
+    override fun sendReplayForEvent(
+        isCrashed: Boolean,
+        eventId: String?,
+        hint: Hint?,
+        onSegmentSent: () -> Unit
+    ) {
+        val sampled = random.sample(options.experimental.sessionReplay.errorSampleRate)
+
+        if (!sampled) {
+            options.logger.log(INFO, "Replay wasn't sampled by errorSampleRate, not capturing for event %s", eventId)
+            return
+        }
+
+        val errorReplayDuration = options.experimental.sessionReplay.errorReplayDuration
+        val now = dateProvider.currentTimeMillis
+        val currentSegmentTimestamp = if (cache?.frames?.isNotEmpty() == true) {
+            // in buffer mode we have to set the timestamp of the first frame as the actual start
+            DateUtils.getDateTime(cache!!.frames.first().timestamp)
+        } else {
+            DateUtils.getDateTime(now - errorReplayDuration)
+        }
+        val segmentId = currentSegment.get()
+        val replayId = currentReplayId.get()
+        val height = recorderConfig.recordingHeight
+        val width = recorderConfig.recordingWidth
+        replayExecutor.submitSafely(options, "$TAG.send_replay_for_event") {
+            var bufferedSegment = bufferedSegments.removeFirstOrNull()
+            while (bufferedSegment != null) {
+                // capture without hint, so the buffered segments don't trigger flush notification
+                bufferedSegment.capture(hub)
+                bufferedSegment = bufferedSegments.removeFirstOrNull()
+                Thread.sleep(100L)
+            }
+            val segment =
+                createSegment(
+                    now - currentSegmentTimestamp.time,
+                    currentSegmentTimestamp,
+                    replayId,
+                    segmentId,
+                    height,
+                    width,
+                    BUFFER
+                )
+            if (segment is ReplaySegment.Created) {
+                segment.capture(hub, hint ?: Hint())
+
+                // we only want to increment segment_id in the case of success, but currentSegment
+                // might be irrelevant since we changed strategies, so in the callback we increment
+                // it on the new strategy already
+                onSegmentSent()
+            }
+        }
+    }
```

Comment:
Is there a plan to use eventId here? AFAICT, event-replay mapping is done via trace context rather than any field. The argument here is used just for logging